setup.py

```diff
@@ -160,17 +156,27 @@ def run(self):
                         doc.write(Archiver.helptext[topic])
                 else:
                     params = {"command": command,
-                              "underline": '-' * len('borg ' + command)}
+                              "underline": '-' * len('borg ' + command),
+                              "description": parser.description,
+                    }
                     doc.write(".. _borg_{command}:\n\n".format(**params))
-                    doc.write("borg {command}\n{underline}\n::\n\n".format(**params))
-                    epilog = parser.epilog
-                    parser.epilog = None
-                    doc.write(re.sub("^", "    ", parser.format_help(), flags=re.M))
+                    doc.write("borg {command}\n{underline}\n\n{description}\n\nSynopsis\n~~~~~~~~\n\n::\n\n".format(**params))
+                    doc.write(re.sub("^", "    ", parser.format_usage().replace('usage: ', '', 1), flags=re.M))
+                    formatter = parser._get_formatter()
+                    # positionals, optionals and user-defined groups
+                    for action_group in parser._action_groups:
+                        if len(action_group._group_actions) > 0:
+                            formatter.start_section("XXX%s\nXXX%s\nXXX\nXXX:" % (action_group.title, '~' * len(action_group.title)))
```

Comment:
i was running out of ideas here... it's kind of ugly, but it's the only way i found of adding headers that wouldn't get indented later one...